### ISSUE TITLE
[ENH] Adapt Move-Split-Merge distance for multivariate time series

### ISSUE THREAD

**Issue by TonyBagnall** [alan-turing-institute/sktime#2480]
**Is your feature request related to a problem? Please describe.**
The Move-Split-Merge (MSM) distance function 
https://ieeexplore.ieee.org/document/6189346
is the top performing distance function for both 1NN classification and kmeans clustering of time series, significantly outperforming the better known dynamic time warping in both situations. The current sktime implementation
https://github.com/alan-turing-institute/sktime/blob/main/sktime/distances/_msm.py
cannot currently handle multivariate input. It would be great if someone could adapt it. If someone does this, there could be the option to collaborate on its evaluation for multivariate time series classification and clustering, which may lead to a publication. 

**Describe the solution you'd like**
The reason MSM is not adapted to multivariate is that it is not immediately obvious how to do it. Our preferred solution would be an implementation of the method described by Ahmed Shifaz
https://arxiv.org/abs/2102.10231

**Additional context**
Note that this needs to be implemented in numba.

**Comment by fkiraly** [comment]
Isn't the "obvious" approach one of reduction/composition, where you just add up the distance from the different variables?

**Comment by TonyBagnall** [comment]
@fkiraly I suggest you read Ahmed's paper, the mutlivariate issue is covered well in that

**Comment by fkiraly** [comment]
> @fkiraly I suggest you read Ahmed's paper, the mutlivariate issue is covered well in that

What I just said, i.e., the reduction approach by summing, is literally the first approach covered in section 3, definition 5 (in a generalization where the p-norm is used, you get summing for p=1)

The natural implementation in sktime, of this, is a compositor that takes univariate distances and sums them up, as said. We can even add a parameter `p` for the p-norm, to bring it in line with definition 5!

**Comment by TonyBagnall** [comment]
yes obviously you an compose the for the independent version, we want the dependent version which is available with all the others

**Comment by fkiraly** [comment]
my comment is just, should we not start with the "simplest" variants, like the compositor?
This would be always sth you will have to compare against in proper benchmarks etc

**Comment by TonyBagnall** [comment]
Your comment may be just, but it's not what I've asked for in this issue.
